src/components/Menu/MenuItem.tsx

```diff
@@ -153,18 +225,69 @@ class MenuItem extends UIComponent<Extendable<MenuItemProps>, MenuItemState> {
           ...accessibility.keyHandlers.root,
         },
         overrideProps: () => ({
-          children: menuItemInner,
+          children: (
+            <>
+              {menuItemInner}
+              {maybeSubmenu}
+            </>
+          ),
+          onClick: this.handleClick,
+          onBlur: this.handleWrapperBlur,
         }),
       })
     }
     return menuItemInner
   }
 
+  private handleWrapperBlur = e => {
+    if (!this.props.inSubmenu && !e.currentTarget.contains(e.relatedTarget)) {
+      this.setState({ submenuOpen: false })
+    }
+  }
+
   protected actionHandlers: AccessibilityActionHandlers = {
     performClick: event => this.handleClick(event),
+    openSubmenu: event => this.openSubmenu(event),
+    closeMenu: event => this.closeMenu(event),
+    closeSubmenu: event => this.closeSubmenu(event),
+  }
+
+  private updateOutsideClickSubscription() {
+    this.outsideClickSubscription.unsubscribe()
+
+    if (this.props.menu && this.state.submenuOpen) {
+      setTimeout(() => {
```

Comment:
note, this approach (presumably, taken from the Popup) has one subtle problem: although it solves the issue of item being opened and closed within the same frame, it also opens possibility for the following behavior:

- MenuItem has opened submenu, this opened prop value doesn't change anyhow for the next JS frame.

///------ JS FRAME --------////
- render method was called (but `submenuOpen ` wasn't changed changed anyhow, it is still `true`)
- updateOutsideClickSubscription
  -  this.outsideClickSubscription.unsubscribe(), no subscription on outside click (!!)
  - setTimeout(() => subscription will be made on the next frame)

- ... some other logic ..
   - note, there is no click subscription at this moment, but it should be (!!)

////---------------------------////

Would suggest to leave a fix for a separate PR, for both components.

Reply:
Will create issue for this, after the PR will be merged